rotkehlchen/globaldb/updates.py

```diff
@@ -300,74 +306,230 @@ def _parse_full_insert(self, insert_text: str) -> AssetData:
             protocol=protocol,
         )
 
-    def _apply_single_version_update(
+    def _process_asset_collection(
             self,
-            connection: DBConnection,
-            version: int,
-            text: str,
-            conflicts: Optional[dict[Asset, Literal['remote', 'local']]],
+            connection: 'DBConnection',
+            action: str,
+            full_insert: str,
+            collection_match: re.Match[str],
     ) -> None:
-        lines = text.splitlines()
-        for action, full_insert in zip(*[iter(lines)] * 2):
-            if full_insert == '*':
-                full_insert = action
+        """Process the insertion of a new asset_collection"""
+        groups = collection_match.groups()
+        if len(groups) != 3:
+            raise DeserializationError(
+                f'At asset DB update could not parse asset collection data out of {action}',
+            )
 
+        try:
+            with connection.savepoint_ctx() as cursor:
+                cursor.execute(action)
+        except sqlite3.Error:
             try:
-                remote_asset_data = self._parse_full_insert(full_insert)
-            except DeserializationError as e:
+                with connection.savepoint_ctx() as cursor:
+                    cursor.execute(full_insert)
+            except sqlite3.Error as e:
                 self.msg_aggregator.add_warning(
-                    f'Skipping entry during assets update to v{version} due '
-                    f'to a deserialization error. {str(e)}',
+                    f'Failed to edit or add asset collection with name {groups[1]} and id '
+                    f'{groups[0]}. Error: {str(e)}',
                 )
-                continue
 
-            local_asset: Optional[Asset] = None
-            with suppress(UnknownAsset):
-                local_asset = Asset(remote_asset_data.identifier).check_existence()
+    def _process_multiasset_mapping(
+            self,
+            connection: 'DBConnection',
+            action: str,
+            full_insert: str,
+            match: re.Match[str],
+    ) -> None:
+        """
+        Process the insertion of a new asset_collection mapping
+        May raise:
+        - DeserializationError
+        - UnknownAsset
+        """
+        groups = match.groups()
+        if len(groups) != 2:
+            raise DeserializationError(
+                f'At asset DB update could not parse asset collection data out of {action}',
+            )
 
+        # check that the asset exists and so does the collection
+        Asset(groups[1]).check_existence()
+        with connection.cursor() as cursor:
```

Comment:
nitpick: Due to the way we have written the code it's equivalent (right now), but we want to be explicit and see it's a read only cursor
```suggestion
        with connection.read_ctx() as cursor:
```